packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -214,7 +214,9 @@ abstract class Route<T> {
   @mustCallSuper
   TickerFuture didPush() {
     return TickerFuture.complete()..then<void>((void _) {
-      navigator?.focusScopeNode.requestFocus();
+      if (navigator?.widget.requestFocus == true) {
```

Comment:
```suggestion
      if (navigator?.widget?.requestFocus == true) {
```

Reply:
Static analyzer tells me that the second `?` is unneeded as the entire clause will short-circuit if `navigator` is null.

Reply:
TIL, lgtm then

Reply:
^ Same XD